### PR TITLE
Expanded Join/SignUp component layout options

### LINKED ISSUE #3123: Allow Sign Up (`join.jsx`) component to render in 2-column layout
Related to #2994 

Allow `join.jsx` component to have the option to render in 2-column layout.

image

### PR DESCRIPTION
**Pull request by mmmavis**
Closes #3123

See screencap of my testing (ignore the turquoise boxes). Note that the alignment of the privacy checkbox and text is a little off. I tried to fix it but couldn't get it solved in 5 mins due to complicated existing CSS overrides + Bootstrap classes. Since this issue can be found on prod, I can file a separate ticket to address this alignment bug.

Bannered campaign test page: https://foundation-mofostaging-pr-3124.herokuapp.com/en/campaigns/bannered-campaign-page-sign/

Screenshot_2019-05-10 about

### CODE REVIEW

File: source/js/components/join/join.jsx [hunk 1]
```diff
@@ -252,7 +252,7 @@ export default class JoinUs extends React.Component {
    * Render the submit button in signup CTA.
    */
   renderSubmitButton() {
-    return <button className="btn btn-primary join-btn w-100">Sign up</button>;
+    return <button className="btn btn-primary w-100">Sign up</button>;
```
Comment (mmmavis) [inline source/js/components/join/join.jsx]: removed as `join-btn` was not being used anywhere in the codebase

File: source/js/components/join/join.jsx [hunk 2]
```diff
@@ -266,22 +266,21 @@ export default class JoinUs extends React.Component {
...
-      <div className="col-12">
-        <form onSubmit={evt => this.processFormData(evt)} className={formClass}>
-          <div className={fieldsWrapperClass}>
-            {this.renderEmailField()}
-            {this.renderPrivacyField()}
-          </div>
-          <div className={submitWrapperClass}>{this.renderSubmitButton()}</div>
-        </form>
-      </div>
+      <form onSubmit={evt => this.processFormData(evt)} className={formClass}>
```
Comment (mmmavis) [inline source/js/components/join/join.jsx]: removed wrapper div

**Comment by kristinashu** [comment]
This is great Mavis! It's nice to see all the variations together (cc @sabrinang @natalieworth @beccaklam just so they are aware).

I assume they all look the same on mobile?

**Comment by mmmavis** [comment]
@kristinashu good point about the mobile! I assume on mobile we want them to all appear in 1-column? If so I'll update my PR and attach another screencap for mobile.

**Comment by kristinashu** [comment]
Yes, one column is great. The two I can see on staging look good and it would be ideal if the others looked that way on mobile too.

**Comment by youriwims** [comment]
@mmmavis can you also provide the link for the use cases in your screenshots?

**Comment by mmmavis** [comment]
@youriwims yup yup see ticket description. I included a link to bannered campaign page there. 👍  For the 2-column layout version there's no live example for it yet as this version will be used in a

**Comment by mmmavis** [comment]
I still need to update this PR. Will comment here again once it's ready for review!

**Comment by mmmavis** [comment]
@kristinashu @youriwims this is ready for review again.

To see all versions live see 
- https://foundation-mofostaging-pr-3124.herokuapp.com/en/styleguide/
- https://foundation-mofostaging-pr-3124.herokuapp.com/en/campaigns/bannered-campaign-page-sign/

**Pull request merged**